File: src/main/java/model/FruitStand.java
package model;

import java.util.Set;

public class FruitStand {

    private Set<Fruit> fruitSet;
    private int standNumber;

    public FruitStand(Set<Fruit> fruitSet, int standNumber) {
        this.fruitSet = fruitSet;
        this.standNumber = standNumber;
    }

    public boolean containsPear() {
        return fruitSet.stream().anyMatch(fruit -> fruit.getName().equals(FruitName.PEAR.getFruitType()));
    }

    public boolean containsTwoFruits() {
        return fruitSet.size() == 2;
    }

    public int getStandNumber() {
        return standNumber;
    }

    public double getTotalPrice() {
        return fruitSet.stream().mapToDouble(Fruit::getPrice).sum();
    }
}

File: src/main/java/model/FruitStand.kt
package model

class FruitStand {
    private var fruitSet: Set<Fruit>? = null
    private var standNumber = 0

    fun FruitStand(fruitSet: Set<Fruit>?, standNumber: Int) {
        this.fruitSet = fruitSet
        this.standNumber = standNumber
    }

    fun containsPear(): Boolean {
        return fruitSet!!.stream().anyMatch { fruit: Fruit -> fruit.getName() == FruitName.PEAR.getFruitType() }
    }

    fun containsTwoFruits(): Boolean {
        return fruitSet!!.size == 2
    }

    fun getFruitSet(): Set<Fruit>? {
        return fruitSet
    }

    fun getStandNumber(): Int {
        return standNumber
    }

    fun getTotalPrice(): Double {
        return fruitSet!!.stream().mapToDouble { obj: Fruit -> obj.getPrice() }.sum()
    }
}
